src/app/shared/services/course.service.ts

```diff
@@ -0,0 +1,64 @@
+import { Injectable } from '@angular/core';
+import { Course } from '../interfaces/course.interface.';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CourseService {
+
+  constructor() { }
+
+  courses: Course[] = [{
+    id: 1,
+    title: 'Some Title 1',
+    creationDate: new Date(2023, 5, 1),
+    duration: 200,
+    description: "Some description 1",
+    topRated: false
+  },
+  {
+    id: 2,
+    title: 'Some Title 2',
+    creationDate: new Date(2023, 5, 30),
+    duration: 100,
+    description: "Some description 2",
+    topRated: false
+  },
+  {
+    id: 3,
+    title: 'Some Title 3',
+    creationDate: new Date(2023, 2, 1),
+    duration: 50,
+    description: "Some description 3",
+    topRated: true
+  }]
+
+  getListCourses(): Course[] {
+    return this.courses;
+  }
+
+  createCourse(course: Course): void {
+    this.courses.push(course);
+  }
+
+  getCourseById(id: number): Course | undefined {
+    const result = this.courses.find(course => course.id === id);
```

Comment:
Can be `return this.courses.find(course => course.id === id);`

Reply:
agreed, thanks